aep/0134.yaml

```diff
@@ -0,0 +1,123 @@
+aliases:
+  UpdateOperation:
+    description: An Update method is a patch on path that ends in a path parameter
+    targets:
+      - formats: ['oas2', 'oas3']
+        given:
+          - $.paths[?(@property.match(/\}$/))].patch
+
+rules:
+  aep-134-content-type:
+    description: The request body content type should be "application/merge-patch+json"
+    severity: warn
+    formats: ['oas3']
+    given: '#UpdateOperation'
+    then:
+      field: requestBody.content.application/merge-patch+json
+      function: truthy
+
+  aep-134-operation-id:
+    description: The operation ID should be Update{resource-singular}.
+    message: The operation ID does not conform to AEP-134
+    severity: error
+    formats: ['oas2', 'oas3']
+    given: '#UpdateOperation'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          properties:
+            operationId:
+              type: string
+              pattern: '^[Uu][Pp][Dd][Aa][Tt][Ee][A-Z].*$'
+          required: ['operationId']
+
+  aep-134-request-body:
+    description: A standard update method must accept the resource in the request body.
+    message: The request body is not an AEP resource.
+    severity: error
+    formats: ['oas3']
+    given: '#UpdateOperation'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          required: ['requestBody']
+          properties:
+            requestBody:
+              # I tried putting this schema is a separate file and $ref'ing it, but it didn't work
```

Comment:
hm... maybe a custom function would work then? https://docs.stoplight.io/docs/spectral/a781e290eb9f9-custom-functions

Reply:
Worth exploring. I opened #24 to track this.